specification/customer-insights/resource-manager/Microsoft.CustomerInsights/2017-04-26/customer-insights.json

```diff
@@ -3149,6 +3149,378 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/predictions/{predictionName}": {
+      "put": {
+        "tags": [
+          "Predictions"
+        ],
+        "operationId": "Predictions_CreateOrUpdate",
+        "x-ms-examples": {
+          "Predictions_CreateOrUpdate": {
+            "$ref": "./examples/PredictionsCreateOrUpdate.json"
+          }
+        },
+        "description": "Creates a Prediction or updates an existing Prediction in the hub.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "predictionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Prediction.",
+            "pattern": "^[a-zA-Z][a-zA-Z0-9_]+$",
+            "minLength": 1,
+            "maxLength": 512
```

Comment:
For operations other than create, e.g. get/delete, I am thinking that maybe not necessary to do name validation. Assume the name pattern changed at some point, there will be legacy data in the system,  we may still want be able to get/delete them, although it's not very likely to happen... What do you think?